### PR TITLE
Run CircleCi with BUILD=debug

### PR DESCRIPTION
**Pull request by wilzbach**
From @jacob-carlborg on Slack:

> I’ve noticed that when compiling DMD in debug mode and running the test the following assertion fails: https://github.com/dlang/dmd/blob/master/src/dmd/dmangle.d#L587
> the assertion fails due to one of characters in the mangled name contains a `-`. Is it the assertion that needs to be updated or is the mangling wrong?
> since we’re not testing the compiler in debug mode, as far as I understand, and everything seems to be working I’m guessing the assertion needs to be updated
> feels slightly pointless to me with the debug code if it’s never run

This just opens the PR with `BUILD=debug`. We still need to fix the error in debug mode, but this can be done independently.

### CODE REVIEW

**Comment by wilzbach** [comment]
"Good news - the testsuite is really failing:

```
 ... runnable/test2.sh
core.exception.AssertError@dmd/dmangle.d(587): Assertion failure
----------------
src/core/exception.d:437 onAssertError [0x857548]
src/core/exception.d:633 _d_assertp [0x8462f2]
dmd/dmangle.d:587 _ZN7Mangler5visitEP11Declaration [0x56e5a0]
dmd/dmangle.d:685 _ZN7Mangler11mangleExactEP15FuncDeclaration [0x56e935]
dmd/dmangle.d:1101 _Z11mangleExactP15FuncDeclaration [0x5702e2]
dmd/tocsym.d:304 _ZN8toSymbol8ToSymbol5visitEP15FuncDeclaration [0x6d791e]
dmd/parsetimevisitor.d:53 _ZN16ParseTimeVisitorI10ASTCodegenE5visitEP19UnitTestDeclaration [0x6a9603]
dmd/func.d:3508 _ZN19UnitTestDeclaration6acceptEP7Visitor [0x60c4d7]
dmd/tocsym.d:444 _Z8toSymbolP7Dsymbol [0x6d6f8f]
dmd/glue.d:818 _Z25FuncDeclaration_toObjFileP15FuncDeclarationb [0x6cf1cd]
dmd/toobj.d:311 _ZN9toObjFile9ToObjFile5visitEP15FuncDeclaration [0x6e233a]
dmd/parsetimevisitor.d:53 _ZN16ParseTimeVisitorI10ASTCodegenE5visitEP19UnitTestDeclaration [0x6a9603]
dmd/func.d:3508 _ZN19UnitTestDeclaration6acceptEP7Visitor [0x60c4d7]
dmd/toobj.d:1369 _Z9toObjFileP7Dsymbolb [0x6e2277]
dmd/glue.d:411 _Z10genObjFileP6Moduleb [0x6ce50e]
dmd/mars.d:1019 int dmd.mars.tryMain(ulong, const(char)**) [0x646bf4]
dmd/mars.d:1124 _Dmain [0x64768a]
executing with args: 
DMD v2.078.0-beta.1-76-g822e4fa-dirty DEBUG
Makefile:193: recipe for target 'test_results/runnable/test2.sh.out' failed
```

For those not using `#dmd` on Slack, this is due to https://github.com/dlang/dmd/pull/6727

**Comment by wilzbach** [comment]
(rebased)

File: .circleci/run.sh [hunk 1]
```diff
@@ -139,19 +139,19 @@ coverage()
...
-    local build_path=generated/linux/release/$MODEL
+    local build_path=generated/linux/$BUILD/$MODEL
     # `generated` gets cleaned in the next step, so we create another _generated
     # The nested folder hierarchy is needed to conform to those specified in
     # the generate dmd.conf
     mkdir -p _${build_path}
     cp $build_path/dmd _${build_path}/host_dmd
     cp $build_path/dmd.conf _${build_path}
-    make -j$N -C src -f posix.mak MODEL=$MODEL HOST_DMD=../_${build_path}/host_dmd PIC="$PIC" clean
-    make -j$N -C src -f posix.mak MODEL=$MODEL HOST_DMD=../_${build_path}/host_dmd ENABLE_COVERAGE=1 ENABLE_WARNINGS=1 PIC="$PIC"
```
Comment (jacob-carlborg) [inline .circleci/run.sh]: Why remove `ENABLE_WARNINGS`?

**Comment by MoonlightSentinel** [comment]
Can we move forward with this to avoid situations like #10656

**Comment by thewilsonator** [comment]
`cp: cannot stat 'generated/linux/debug/64/dmd.conf': No such file or directory`

Maybe the `.conf` generation also needs to be redirected also?

**Comment by MoonlightSentinel** [comment]
It should already but `build.d` skipped `dmd.conf` for unknown reasons...

**Comment by Geod24** [comment]
Can we get those PRs rebased instead of creating merge commits please ?

**Comment by MoonlightSentinel** [comment]
Agreed but I used the web interface because GitHub previously rejected my rebase complaining about `runnable_cxx.yml`.

**Review by PetarKirov — APPROVED** [review]

File: .circleci/run.sh (hunk 1 above)
Comment (MoonlightSentinel) [reply .circleci/run.sh]: Reverted this change because it's unrelated (although doesn't matter because  warnings are enabled by default)

**Review by Geod24 — APPROVED** [review]

**Pull request merged**